packages/gemini-runner/src/adapters.ts

```diff
@@ -0,0 +1,149 @@
+import type { SDKMessage, SDKUserMessage } from "cyrus-core";
+import type { GeminiMessageEvent, GeminiStreamEvent } from "./types.js";
+
+/**
+ * Convert a Gemini stream event to Claude SDK message format
+ *
+ * This adapter maps Gemini CLI's streaming events to the Claude SDK's SDKMessage
+ * format, allowing GeminiRunner to implement the IAgentRunner interface.
+ *
+ * @param event - Gemini CLI stream event
+ * @param sessionId - Current session ID (may be null initially)
+ * @returns SDKMessage or null if event type doesn't map to a message
+ */
+export function geminiEventToSDKMessage(
+	event: GeminiStreamEvent,
+	sessionId: string | null,
+): SDKMessage | null {
+	switch (event.type) {
+		case "message": {
+			const messageEvent = event as GeminiMessageEvent;
+			if (messageEvent.role === "user") {
+				return {
+					type: "user",
+					message: {
+						role: "user",
+						content: messageEvent.content,
+					},
+					parent_tool_use_id: null,
+					session_id: sessionId || "pending",
+				} satisfies SDKUserMessage;
+			} else {
+				// Assistant message
+				return {
+					type: "assistant",
+					message: {
+						role: "assistant",
+						content: messageEvent.content,
+					},
+					session_id: sessionId || "pending",
+				} as unknown as SDKMessage;
+			}
+		}
+
+		case "init":
+			// Init events don't map directly to messages
+			// Session ID is extracted separately
+			return null;
+
+		case "tool_use":
+			// Map to Claude's tool_use format
+			return {
+				type: "assistant",
+				message: {
+					role: "assistant",
+					content: [
+						{
+							type: "tool_use",
+							id: `${event.tool_name}_${Date.now()}`, // Generate unique ID
+							name: event.tool_name,
+							input: event.parameters,
+						},
+					],
+				},
+				session_id: sessionId || "pending",
+			} as unknown as SDKMessage;
+
+		case "tool_result":
+			// Map to Claude's tool_result format
+			return {
+				type: "user",
+				message: {
+					role: "user",
+					content: [
+						{
+							type: "tool_result",
+							tool_use_id: `${event.tool_name}_${Date.now()}`,
+							content: JSON.stringify(event.result),
+						},
+					],
+				},
+				parent_tool_use_id: null,
+				session_id: sessionId || "pending",
+			} as unknown as SDKMessage;
+
+		case "result":
+			// Final result - map to assistant message with the response
+			if (event.error) {
+				// Error result - log but return null since SDK doesn't have a direct error message type
+				console.error(`[GeminiAdapter] Error result: ${event.error.message}`);
+				return null;
+			} else {
+				// Success result with final response
+				return {
+					type: "assistant",
+					message: {
+						role: "assistant",
+						content: event.response,
+					},
+					session_id: sessionId || "pending",
+				} as unknown as SDKMessage;
```

Comment:
instead of writing this way, and assuming you've presented the right info, how about defining a value which is a SDKUserMessage and defining anobject which satisfies it then we will get the benefits of type checking. Then return it.